src/main/java/au/org/ala/biocache/stream/ProcessDownload.java

```diff
@@ -250,21 +250,24 @@ private void intersectAnalysisLayers() {
                 }
                 if (i < batch.size()) {
                     Reader reader = layersService.sample(headers.analysisIds, points, null);
-
-                    CSVReader csv = new CSVReader(reader);
-                    intersection = csv.readAll();
-                    csv.close();
-
-                    for (int j = 0; j < batch.size(); j++) {
-                        if (j > batch.size()) {
-                            //+1 offset for header row in intersection list
-                            String[] sampling = intersection.get(j + 1);
-                            //+2 offset for latitude,longitude columns in sampling array
-                            if (sampling != null && sampling.length == headers.analysisIds.length + 2) {
-                                // suitable space is already available in each batch row String[]
-                                System.arraycopy(sampling, 2, batch.get(j), headers.labels.length, sampling.length - 2);
+                    if (reader != null) {
+                        CSVReader csv = new CSVReader(reader);
+                        intersection = csv.readAll();
+                        csv.close();
+
+                        for (int j = 0; j < batch.size(); j++) {
+                            if (j > batch.size()) {
```

Comment:
Remove `if` statement that is always false.

Reply:
If spatial service does not return 'completed' status,  Null reader will be returned.